### PR TITLE
Sso transition

### PR DESCRIPTION
**Pull request by jenreiher-bcgov**

### CODE REVIEW

File: docs/use-github-in-bcgov/bc-government-organizations-in-github.md [hunk 1]
```diff
@@ -41,31 +40,109 @@ The `bcgov-c` organization stores temporary (up to 12 months), private repositor
...
+ 
+1. Have any two-factor devices, and personal GitHub login and IDIR login ready for linking 
+2. Log into GitHub and follow the prompts on the pop up to connect your GitHub and IDIR accounts together. [Learn more about how this works](). 
+3. Reauthorize your personal access tokens and/or SSH keys so they continue to work with SSO enabled. [Learn more about how to do this](). 
+ 
+That’s it, your accounts are now linked.  
+ 
+#### For contractors 
+ 
+If you don’t have an IDIR account, [we have guidance on ordering a compatible IDIR] ](faq link). It can take several days to complete depending on your ministry.We recommend doing with this step as soon as possible to avoid access disruptions.
```
Comment (Shivanganii) [inline docs/use-github-in-bcgov/bc-government-organizations-in-github.md:98]: [we have guidance on ordering a compatible IDIR] ](https://bcdevex.atlassian.net/wiki/spaces/DEVEXP/pages/1514274870/GitHub+Single+Sign-On+SSO+Transition+Guide)

File: docs/use-github-in-bcgov/bc-government-organizations-in-github.md [hunk 2]
```diff
@@ -41,31 +40,110 @@ The `bcgov-c` organization stores temporary (up to 12 months), private repositor
...
+Your product team can only have a **permanent**, private repository in `bcgov-c` if it is a GitOps repository with ArgoCD manifests. 
+
+
+## Single sign-on is coming to the B.C. government’s GitHub organizations
+
+We're enabling single sign-on (SSO) for GitHub. We’re starting with the bcgov GitHub organization. After we finish transitioning bcgov, we’ll enable other OCIO-managed GitHub organizations. 
+  
+To keep access to the bcgov GitHub organization, all users must link their IDIRs to their GitHub IDs between **June 30th and July 30th, 2024**. If users don’t complete this by July  30th, 2024, they’ll be automatically removed from the bcgov GitHub organization, and by extension, all the code repositories connected to bcgov.   
+
+Contractors without IDIRs must get one to before they can enable SSO. [Learn how to order an IDIR]([https://bcdevex.atlassian.net/wiki/spaces/DEVEXP/pages/1514274870/GitHub+Single+Sign-On+SSO+Transition+Guide](https://citz-do.atlassian.net/servicedesk/customer/portal/2/article/78053380)).
```
Comment (jenreiher-bcgov) [inline docs/use-github-in-bcgov/bc-government-organizations-in-github.md:52]: @Shivanganii link is twice here - I think you only want the second one

File: docs/use-github-in-bcgov/bc-government-organizations-in-github.md [hunk 3]
```diff
@@ -41,31 +40,110 @@ The `bcgov-c` organization stores temporary (up to 12 months), private repositor
...
+- Access tokens and/or SSH keys that your app or team may rely on that will be disconnected when user access is removed 
+  - IDE or command line tools and/or automation scripts 
+ 
+### The changes 
+ 
+Before this upgrade, users only needed their personal GitHub IDs to access the bcgov GitHub organization as a member. 
+ 
+By July 30, 2024, users must follow prompts in GitHub to link their personal GitHub ID with their IDIR account.  If not, users will be automatically removed. 
+ 
+[Learn more about how linking works]().
```
Comment (jenreiher-bcgov) [inline docs/use-github-in-bcgov/bc-government-organizations-in-github.md:81]: @Shivanganii missing a link here

File: docs/use-github-in-bcgov/bc-government-organizations-in-github.md [hunk 4]
```diff
@@ -41,31 +40,110 @@ The `bcgov-c` organization stores temporary (up to 12 months), private repositor
...
+Ministries won’t be charged for upgraded GitHub licenses.  However, contractors need IDIRs before they can enable SSO. Ministries may need to seek expense approvals to order [IDIRs for contracted resources without one](https://citz-do.atlassian.net/servicedesk/customer/portal/2/article/78053380). 
+  
+### Directions to link your accounts before July 30
+ 
+#### For employees 
+ 
+[placeholder for video to show what to expect]()
+ 
+1. Have any two-factor devices, and personal GitHub login and IDIR login ready for linking 
+2. Log into GitHub and follow the prompts on the pop up to connect your GitHub and IDIR accounts together. [Learn more about how this works]().
```
Comment (jenreiher-bcgov) [inline docs/use-github-in-bcgov/bc-government-organizations-in-github.md:92]: @Shivanganii missing a link here

File: docs/use-github-in-bcgov/bc-government-organizations-in-github.md [hunk 5]
```diff
@@ -41,31 +40,110 @@ The `bcgov-c` organization stores temporary (up to 12 months), private repositor
...
+  
+### Directions to link your accounts before July 30
+ 
+#### For employees 
+ 
+[placeholder for video to show what to expect]()
+ 
+1. Have any two-factor devices, and personal GitHub login and IDIR login ready for linking 
+2. Log into GitHub and follow the prompts on the pop up to connect your GitHub and IDIR accounts together. [Learn more about how this works](). 
+3. Reauthorize your personal access tokens and/or SSH keys so they continue to work with SSO enabled. [Learn more about how to do this]().
```
Comment (jenreiher-bcgov) [inline docs/use-github-in-bcgov/bc-government-organizations-in-github.md:93]: @Shivanganii missing a link here

File: docs/use-github-in-bcgov/bc-government-organizations-in-github.md [hunk 6]
```diff
@@ -41,31 +40,110 @@ The `bcgov-c` organization stores temporary (up to 12 months), private repositor
...
+ 
+1. Have any two-factor devices, and personal GitHub login and IDIR login ready for linking 
+2. Log into GitHub and follow the prompts on the pop up to connect your GitHub and IDIR accounts together. [Learn more about how this works](). 
+3. Reauthorize your personal access tokens and/or SSH keys so they continue to work with SSO enabled. [Learn more about how to do this](). 
+ 
+That’s it, your accounts are now linked.  
+ 
+#### For contractors 
+ 
+If you don’t have an IDIR account, [we have guidance on ordering a compatible IDIR] ](https://citz-do.atlassian.net/servicedesk/customer/portal/2/article/78053380). It can take several days to complete depending on your ministry.We recommend doing with this step as soon as possible to avoid access disruptions.
```
Comment (jenreiher-bcgov) [inline docs/use-github-in-bcgov/bc-government-organizations-in-github.md:99]: @Shivanganii  extra " ]" here

File: docs/use-github-in-bcgov/bc-government-organizations-in-github.md [hunk 7]
```diff
@@ -41,31 +40,110 @@ The `bcgov-c` organization stores temporary (up to 12 months), private repositor
...
+#### For contractors 
+ 
+If you don’t have an IDIR account, [we have guidance on ordering a compatible IDIR] ](https://citz-do.atlassian.net/servicedesk/customer/portal/2/article/78053380). It can take several days to complete depending on your ministry.We recommend doing with this step as soon as possible to avoid access disruptions.  
+ 
+After getting an IDIR, follow the steps below: 
+
+[placeholder for video to show what to expect]()
+
+1. Have any two-factor device, and personal GitHub login and IDIR login ready for linking 
+2. Log into GitHub and follow the prompts on the pop up to connect your GitHub and IDIR accounts together. [Learn more about how this works]().
```
Comment (jenreiher-bcgov) [inline docs/use-github-in-bcgov/bc-government-organizations-in-github.md:106]: @Shivanganii missing a link here

File: docs/use-github-in-bcgov/bc-government-organizations-in-github.md [hunk 8]
```diff
@@ -41,31 +40,110 @@ The `bcgov-c` organization stores temporary (up to 12 months), private repositor
...
+ 
+If you don’t have an IDIR account, [we have guidance on ordering a compatible IDIR] ](https://citz-do.atlassian.net/servicedesk/customer/portal/2/article/78053380). It can take several days to complete depending on your ministry.We recommend doing with this step as soon as possible to avoid access disruptions.  
+ 
+After getting an IDIR, follow the steps below: 
+
+[placeholder for video to show what to expect]()
+
+1. Have any two-factor device, and personal GitHub login and IDIR login ready for linking 
+2. Log into GitHub and follow the prompts on the pop up to connect your GitHub and IDIR accounts together. [Learn more about how this works](). 
+3. If linking fails, your IDIR configuration is not compatible with our GitHub SSO. You must [request an upgrade to a P2 level IDIR account]().
```
Comment (jenreiher-bcgov) [inline docs/use-github-in-bcgov/bc-government-organizations-in-github.md:107]: @Shivanganii missing a link here

File: docs/use-github-in-bcgov/bc-government-organizations-in-github.md [hunk 9]
```diff
@@ -41,31 +40,110 @@ The `bcgov-c` organization stores temporary (up to 12 months), private repositor
...
+If you don’t have an IDIR account, [we have guidance on ordering a compatible IDIR] ](https://citz-do.atlassian.net/servicedesk/customer/portal/2/article/78053380). It can take several days to complete depending on your ministry.We recommend doing with this step as soon as possible to avoid access disruptions.  
+ 
+After getting an IDIR, follow the steps below: 
+
+[placeholder for video to show what to expect]()
+
+1. Have any two-factor device, and personal GitHub login and IDIR login ready for linking 
+2. Log into GitHub and follow the prompts on the pop up to connect your GitHub and IDIR accounts together. [Learn more about how this works](). 
+3. If linking fails, your IDIR configuration is not compatible with our GitHub SSO. You must [request an upgrade to a P2 level IDIR account]().  
+4. Reauthorize your personal access tokens and/or SSH keys so they continue to work with SSO enabled. [Learn more about how to do this]().
```
Comment (jenreiher-bcgov) [inline docs/use-github-in-bcgov/bc-government-organizations-in-github.md:108]: @Shivanganii missing a link here

File: docs/use-github-in-bcgov/bc-government-organizations-in-github.md [hunk 10]
```diff
@@ -41,31 +40,110 @@ The `bcgov-c` organization stores temporary (up to 12 months), private repositor
...
+ 
+That’s it, your accounts are now linked.  
+ 
+### Directions to link your account after July 30 
+
+If you are a contractor without an IDIR, follow the steps in the [IDIR transition guide](https://citz-do.atlassian.net/servicedesk/customer/portal/2/article/78053380) before proceeding. 
+ 
+1. If you have a valid IDIR, ask anyone with access to the organization to re-invite you. [Learn how to request access to the bcgov GitHub organization](https://developer.gov.bc.ca/docs/default/component/bc-developer-guide/use-github-in-bcgov/bc-government-organizations-in-github/#bcgov). 
+2. Once you have access again, your teammates with admin access may need to re-invite you to the repositories you had access to before.  
+3. Reauthorize your personal access tokens and/or SSH keys so they continue to work with SSO enabled. [Learn more about how to do this]().
```
Comment (jenreiher-bcgov) [inline docs/use-github-in-bcgov/bc-government-organizations-in-github.md:118]: @Shivanganii missing a link here

File: docs/use-github-in-bcgov/bc-government-organizations-in-github.md [hunk 11]
```diff
@@ -41,31 +40,110 @@ The `bcgov-c` organization stores temporary (up to 12 months), private repositor
...
+ 
+1. If you have a valid IDIR, ask anyone with access to the organization to re-invite you. [Learn how to request access to the bcgov GitHub organization](https://developer.gov.bc.ca/docs/default/component/bc-developer-guide/use-github-in-bcgov/bc-government-organizations-in-github/#bcgov). 
+2. Once you have access again, your teammates with admin access may need to re-invite you to the repositories you had access to before.  
+3. Reauthorize your personal access tokens and/or SSH keys so they continue to work with SSO enabled. [Learn more about how to do this](). 
+ 
+### Upgrading other GitHub organizations 
+ 
+The bcgov-c, BCDevOps organizations will transition to SSO on different timelines. It will be the same process moving forward for any organization upgrade.  
+ 
+[Sign up to the DevX newsletter]() for updates on these changes.
```
Comment (jenreiher-bcgov) [inline docs/use-github-in-bcgov/bc-government-organizations-in-github.md:124]: @Shivanganii ah shoot, we need to track down how to make a sign up page for Mail Chimp - something for standup!

File: docs/use-github-in-bcgov/bc-government-organizations-in-github.md [hunk 12]
```diff
@@ -41,31 +40,110 @@ The `bcgov-c` organization stores temporary (up to 12 months), private repositor
...
+2. Once you have access again, your teammates with admin access may need to re-invite you to the repositories you had access to before.  
+3. Reauthorize your personal access tokens and/or SSH keys so they continue to work with SSO enabled. [Learn more about how to do this](). 
+ 
+### Upgrading other GitHub organizations 
+ 
+The bcgov-c, BCDevOps organizations will transition to SSO on different timelines. It will be the same process moving forward for any organization upgrade.  
+ 
+[Sign up to the DevX newsletter]() for updates on these changes. 
+ 
+For current [B.C. Government private Enterprise]() organizations, we’ll work with your respective Information Management Branches and Information Systems Branches to identify next steps
```
Comment (jenreiher-bcgov) [inline docs/use-github-in-bcgov/bc-government-organizations-in-github.md:126]: @Shivanganii missing a link here

**Pull request merged**